# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: xpfe/appshell/nsIAppWindow.idl
# Commit: 765d3364cca0
# Full Hash: 765d3364cca0a67c614cb4fb5dff413a8c29d834
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2020-04-07 03:53:09
# Regressor Bug: 1616353
# File Overlap Count: 1
# Description:
#   Bug 1616353 - Part 7.2: Create and use nsOpenWindowInfo types in nsWindowWatcher logic, r=kmag
#   
#   This patch builds on top of part 7.1 by creating this object within
#   nsWindowWatcher and ContentParent to carry the relevant information through
#   provider interfaces when creating new content windows. The nsOpenWindowInfo
# ==============================================================================

diff -r 8a13355b4ac4 -r 765d3364cca0 xpfe/appshell/nsIAppWindow.idl
--- a/xpfe/appshell/nsIAppWindow.idl	Mon Apr 06 14:29:59 2020 +0000
+++ b/xpfe/appshell/nsIAppWindow.idl	Mon Apr 06 14:30:02 2020 +0000
@@ -23,6 +23,7 @@
 interface nsIXULBrowserWindow;
 interface nsIRemoteTab;
 interface mozIDOMWindowProxy;
+interface nsIOpenWindowInfo;
 
 native LiveResizeListenerArray(nsTArray<RefPtr<mozilla::LiveResizeListener>>);
 
@@ -126,17 +127,12 @@
   /**
    * Create a new window.
    * @param aChromeFlags see nsIWebBrowserChrome
-   * @param aOpeningTab the RemoteTab that requested this new window be opened.
-   *                    Can be left null.
-   * @param aOpener The window which is requesting that this new window be opened.
-   * @param aNextRemoteTabId The integer ID of the next remote tab actor to use.
-   *        0 means there is no next remote tab actor to use.
+   * @param aOpenWindowInfo information about the request for a content window
+   *                        to be opened. Will be null for non-content loads.
    * @return the newly minted window
    */
   nsIAppWindow createNewWindow(in int32_t aChromeFlags,
-                               in nsIRemoteTab aOpeningTab,
-                               in mozIDOMWindowProxy aOpener,
-                               in unsigned long long aNextRemoteTabId);
+                               in nsIOpenWindowInfo aOpenWindowInfo);
 
   attribute nsIXULBrowserWindow XULBrowserWindow;
 
@@ -179,9 +175,8 @@
                                                  in int32_t shellItemHeight);
 
   /**
-   * If the window was opened as a content window by script, this will return the
-   * integer ID of the next RemoteTab actor to use.
+   * If the window was opened as a content window, this will return the initial
+   * nsIOpenWindowInfo to use.
    */
-  [noscript]
-  readonly attribute unsigned long long nextRemoteTabId;
+  readonly attribute nsIOpenWindowInfo initialOpenWindowInfo;
 };
